### PR TITLE
[Basic] Avoid repeated hash lookups (NFC)

### PR DESCRIPTION
**Pull request by kazutakahirata**

### CODE REVIEW

File: clang/lib/Basic/TargetID.cpp [hunk 1]
```diff
@@ -92,11 +92,10 @@ parseTargetIDWithFormatCheckingOnly(llvm::StringRef TargetID,
     if (Sign != '+' && Sign != '-')
       return std::nullopt;
     bool IsOn = Sign == '+';
-    auto Loc = FeatureMap->find(Feature);
+    auto [Loc, Inserted] = FeatureMap->try_emplace(Feature, IsOn);
```
Comment (nikic) [inline clang/lib/Basic/TargetID.cpp:95]: Loc not used?

Comment (kazutakahirata) [reply clang/lib/Basic/TargetID.cpp]: Thank you for pointing this out!

**Comment by kazutakahirata** [comment]
Please take a look.  Thanks!

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**